hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -203,6 +198,43 @@ public void run(Operation op) {
         }
     }
 
+    private void call(Operation op) throws Exception {
+        CallStatus callStatus = op.call();
+
+        switch (callStatus) {
+            case NO_RESPONSE:
+                // todo: currently there is no difference between NO_RESPONSE and OFFLOADED
+                op.afterRun();
+                break;
+            case OFFLOADED:
+                op.afterRun();
+                break;
```

Comment:
Do we want to run the afterRun here? not after the result is returned?

Reply:
Good point: Currently we are free to choose.

It makes sense to execute it since the run/call was executed, so I would assume the afterRun would then be executed as well.

However.. in case of a true continuation, the run method could be called many times, so the afterRun could be called many times. If e,g, cleanup or statistics updating would be done in the afterRun, this would be executed more than once which can lead to all kinds of problems.

So afterRun has become ambiguous.